volume/devicemapper/devicemapper.go

```diff
@@ -677,6 +716,14 @@ func (d *DeviceMapperDriver) ensureInitialized() error {
 	return nil
 }
 
+func (d *DeviceMapperDriver) deleteDevice(deviceName string, syncDelete bool) error {
+	if deviceName == "base" || deviceName == "" {
```

Comment:
How we handle that depends on if this was a snapshot device, a temporary device, or a volume's main device.  In Right now that is handled by the methods that call this one.